assets_test.go

```diff
@@ -30,3 +30,101 @@ func TestAsset(t *testing.T) {
 	require.Equal(t, "GAJVD2WOS7QXLSGFUQ3VIDEFG5I7S3VWL4X3V5FEFN4N2OC5CQDMHHZS", summary.AssetIssuer)
 	require.Empty(t, summary.UnverifiedWellKnownLink)
 }
+
+func TestAssetSearch(t *testing.T) {
+	helper, client, network := testclient.Setup(t)
+	SetClientAndNetwork(client, network)
+	helper.SetState(t, "assetSearch")
+
+	var res []AssetSummary
+	var expectedMatch AssetSummary
+	var err error
+	search := func(code, issuer string) ([]AssetSummary, error) {
+		arg := AssetSearchArg{
+			AssetCode: code,
+			IssuerID:  issuer,
+		}
+		return AssetSearch(arg)
+	}
+
+	// finds an assetcode with a bunch of issuers
+	res, err = search("BTC", "")
+	require.NoError(t, err)
+	expectedMatch = AssetSummary{
+		UnverifiedWellKnownLink: "",
+		AssetType:               "credit_alphanum4",
+		AssetCode:               "BTC",
+		AssetIssuer:             "GA5FUT6R7MD6CXX6T42HJ6E7NYGNIWAQKWMWRUYR42YLRQG3YWUNRNZU",
+		Amount:                  "0.0049880",
+		NumAccounts:             2,
+	}
+	require.Equal(t, res[0], expectedMatch)
+	require.Equal(t, len(res), 10)
+
+	// finds an issuer with a bunch of assets
+	res, err = search("", "GAJVD2WOS7QXLSGFUQ3VIDEFG5I7S3VWL4X3V5FEFN4N2OC5CQDMHHZS")
+	require.NoError(t, err)
+	expectedMatch = AssetSummary{
+		UnverifiedWellKnownLink: "",
+		AssetType:               "credit_alphanum4",
+		AssetCode:               "GBP",
+		AssetIssuer:             "GAJVD2WOS7QXLSGFUQ3VIDEFG5I7S3VWL4X3V5FEFN4N2OC5CQDMHHZS",
+		Amount:                  "3000.0000000",
+		NumAccounts:             3,
+	}
+	var foundMatch bool
+	for _, asset := range res {
+		if asset == expectedMatch {
+			foundMatch = true
+			break
+		}
+	}
+	require.True(t, foundMatch)
+	require.Equal(t, len(res), 10)
+
+	// finds an exact match
+	res, err = search("BTC", "GA5FUT6R7MD6CXX6T42HJ6E7NYGNIWAQKWMWRUYR42YLRQG3YWUNRNZU")
+	require.NoError(t, err)
+	expectedMatch = AssetSummary{
+		UnverifiedWellKnownLink: "",
+		AssetType:               "credit_alphanum4",
+		AssetCode:               "BTC",
+		AssetIssuer:             "GA5FUT6R7MD6CXX6T42HJ6E7NYGNIWAQKWMWRUYR42YLRQG3YWUNRNZU",
+		Amount:                  "0.0049880",
+		NumAccounts:             2,
+	}
+	foundMatch = false
+	for _, asset := range res {
+		if asset == expectedMatch {
+			foundMatch = true
+			break
+		}
+	}
+	require.True(t, foundMatch)
+	require.Equal(t, len(res), 1)
+
```

Comment:
can you add a test for credit_alphanum12 asset as well?

Reply:
done